### ISSUE TITLE
Glumpy + Mac, SDL2 stuck in OpenGL 2.1, Pyglet stuck in 4.1.

### ISSUE THREAD

**Issue by atalkingegg** [glumpy/glumpy#317]
After an hour trying to get GLSL 410 shaders to compile, I finally tried adding some code:
Try410 = True
if Try410:
    app.use("sdl2")
    config = app.configuration.Configuration()
    config.major_version = 4
    config.minor_version = 1
    config.profile = "core"

Then in the on_draw function,
    version = gl.glGetString(gl.GL_VERSION).decode('utf-8')
    shading_language_version = gl.glGetString(gl.GL_SHADING_LANGUAGE_VERSION).decode('utf-8')
    print("OpenGL Version:", version)
    print("Shading Language Version:", shading_language_version)

This results in the output:
OpenGL Version: 2.1 NVIDIA-10.33.0 387.10.10.10.40.140
Shading Language Version: 1.20

It looks like setting the configuration for SDL2 to get a different OpenGL version is ignored. I've seen that loading Pyglet gets OpenGL Version 4.1 by default, but can't load any shaders. Anyone on a mac successfully getting onto a later OpenGL version where it's useful? Thanks!

**Comment by rougier** [comment]
Sorry for late answer. What do you get from running [app-config.py](https://github.com/glumpy/glumpy/blob/master/examples/app-config.py)?

**Comment by atalkingegg** [comment]
Hi Nicolas,
Having not seen any activity on this project, I've moved on through ten
other things since. Whatever I've posted is what's there.
If memory serves, I ran across a pdf book that was both very good, and also
very badly formatted, the iBook reader and preview both see it as one very
very long page, so moving around is a royal pain, and rendering each
section is very slow on both mac and iphone.
That said, the content, which again if memory serves me correctly, was
doing opengl for scientific visualization in python, which is a really nice
cross of three things I already have lots of background in but separately.
Again well written, but getting a dozen or so examples in and hitting too
many little fixable but annoying bugs, and seeing no active community, I
figured it was another cool but dead project on the internet, but a waste
of time to put any more effort into, so I moved on.

If you are going to revisit this project and bring it back to life, I'd be
happy to dig out and share my updated code examples, and find you some new
bugs to fix as I go. I've got another "almost opengl" project that I'm
working around, and was thinking that python glumpy could be a nice and fun
way to get a bunch of independent working opengl modules built in a
familiar and flexible environment. I looked at jupyter notebooks mixed with
glumpy or other GL-like graphics options, but didn't get very far. WebGL
was another thing but not in Python and didn't solve anything for me. I've
done a fair bit of work with SDL2, but that's all C/C++, and tends to use
old-style opengl syntax. My other project target is Raspberry Pi's and
older intel macs, so OpenGL2.1 and GLSL 1.2 is about the limit.
Let me know.
John G.





If you are
